bootstrap/bootstrap.go

```diff
@@ -293,11 +293,56 @@ func (b *Bootstrap) executeHook(ctx context.Context, hookCfg HookConfig) error {
 
 	b.shell.Headerf("Running %s hook", hookName)
 
+	isBinary, err := shellscript.IsBinaryExecutable(hookCfg.Path)
+	if err != nil {
+		return fmt.Errorf("checking if %s hook is a binary: %w", hookName, err)
+	}
+
+	sb, err := shellscript.ShebangLine(hookCfg.Path)
+	if err != nil {
+		return fmt.Errorf("reading shebang line for %s hook: %w", hookName, err)
+	}
+
+	switch {
```

Comment:
This looks fairly robust to me, but I wonder: do we want to give the user the ability (say, with a flag) to always run unwrapped?

Reply:
On reflection, it's probably likely to just break people for minimal gain.